src/main/java/uk/gov/companieshouse/api/testdata/config/MongoConfig.java

```diff
@@ -74,14 +73,14 @@ public CompanyPscStatementRepository companyPscStatement() {
     }
 
     private MongoTemplate createMongoTemplate(final String database) {
-        SimpleMongoClientDbFactory simpleMongoDbFactory = new SimpleMongoClientDbFactory(
+        SimpleMongoClientDatabaseFactory simpleMongoDbFactory = new SimpleMongoClientDatabaseFactory(
                 MongoClients.create(this.mongoProperties.getUri()), database);
         MappingMongoConverter mappingMongoConverter = getMappingMongoConverter(simpleMongoDbFactory);
         return new MongoTemplate(simpleMongoDbFactory, mappingMongoConverter);
     }
 
-    private MappingMongoConverter getMappingMongoConverter(MongoDbFactory factory) {
-        DbRefResolver dbRefResolver = new DefaultDbRefResolver(factory);
+    private MappingMongoConverter getMappingMongoConverter(SimpleMongoClientDatabaseFactory simpleMongoDbFactory) {
```

Comment:
I wonder if this could use the `MongoDatabaseFactory` interface rather than a specific implementation? 
Also the parameter name could still be `factory`?